WordPressAuthenticator/GoogleSignIn/NewGoogleAuthenticator.swift

```diff
@@ -0,0 +1,121 @@
+import AuthenticationServices
+
+public class NewGoogleAuthenticator: NSObject {
+
+    let clientId: GoogleClientId
+    let scheme: String
+    let audience: String
+    let contextProvider: ASWebAuthenticationPresentationContextProviding
+
+    let oauthTokenGetter: GoogleOAuthTokenGetting
+
+    public convenience init(
+        clientId: GoogleClientId,
+        scheme: String,
+        audience: String,
+        viewController: UIViewController,
+        urlSession: URLSession
+    ) {
+        self.init(
+            clientId: clientId,
+            scheme: scheme,
+            audience: audience,
+            contextProvider: WebAuthenticationPresentationContext(viewController: viewController),
+            oautTokenGetter: GoogleOAuthTokenGetter(dataGetter: urlSession)
+        )
+    }
+
+    public convenience init(
+        clientId: GoogleClientId,
+        scheme: String,
+        audience: String,
+        contextProvider: ASWebAuthenticationPresentationContextProviding,
+        urlSession: URLSession
+    ) {
+        self.init(
+            clientId: clientId,
+            scheme: scheme,
+            audience: audience,
+            contextProvider: contextProvider,
+            oautTokenGetter: GoogleOAuthTokenGetter(dataGetter: urlSession)
+        )
+    }
+
+    init(
+        clientId: GoogleClientId,
+        scheme: String,
+        audience: String,
+        contextProvider: ASWebAuthenticationPresentationContextProviding,
+        oautTokenGetter: GoogleOAuthTokenGetting
+    ) {
+        self.clientId = clientId
+        self.scheme = scheme
+        self.audience = audience
+        self.oauthTokenGetter = oautTokenGetter
+        self.contextProvider = contextProvider
+    }
+
+    /// Get the user's OAuth token from their Google account. This token can be used to authenticate with the WordPress backend.
+    public func getOAuthToken() async throws -> IDToken {
```

Comment:
I'm wondering if it'd be better to move the `viewController` initializer argument to here as a function argument. It appears we don't need to hold on to the view controller instance across the whole life of the `NewGoogleAuthenticator` instance? And something like `googleAuthenticator.getOAuthToken(from: viewController)` seems like better indicating that a view controller will be used to present the authentication flow.

Reply:
That's along what I was thinking here:

https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/d1b6c91f5ac76a5456da669b7d74bc6d145c01af/WordPressAuthenticator/Unified%20Auth/GoogleAuthenticator.swift#L310-L322

If that's okay with you, I'd like to tackle it in a dedicated PR

Reply:
Addressed in #755